actpool/actqueue.go

```diff
@@ -31,10 +32,15 @@ type noncePriorityQueue []*nonceWithTTL
 
 func (h noncePriorityQueue) Len() int           { return len(h) }
 func (h noncePriorityQueue) Less(i, j int) bool { return h[i].nonce < h[j].nonce }
-func (h noncePriorityQueue) Swap(i, j int)      { h[i], h[j] = h[j], h[i] }
+func (h noncePriorityQueue) Swap(i, j int) {
+	h[i], h[j] = h[j], h[i]
+	h[i].idx = i
+	h[j].idx = j
+}
 
 func (h *noncePriorityQueue) Push(x interface{}) {
 	if in, ok := x.(*nonceWithTTL); ok {
+		in.idx = h.Len()
```

Comment:
`in.idx = len(*h)` Don't use public func inside another public func.

Reply:
ok, fixed